### PR TITLE
use apiserver args for enabling and disabling alpha versions

### PR DESCRIPTION
**Pull request by deads2k**
This allows us to disable alpha versions by default, but still allow them to be enabled via config.

@mfojtik

### CODE REVIEW

**Comment by mfojtik** [comment]
[test]

File: pkg/cmd/server/api/helpers.go [hunk 1]
```diff
@@ -593,7 +594,34 @@ func HasOpenShiftAPILevel(config MasterConfig, apiLevel string) bool {
 	return apiLevelSet.Has(apiLevel)
 }
 
-// GetEnabledAPIVersionsForGroup returns the list of API Versions that are enabled for that group
+const kubeAPIEnablementFlag = "runtime-config"
+
+// GetKubeAPIServerFlagAPIEnablement parses the available flag at the groupVersion level
+// with no support for individual resources and no support for the legacy API.
+func GetKubeAPIServerFlagAPIEnablement(flagValue []string) map[schema.GroupVersion]bool {
```
Comment (mfojtik) [inline pkg/cmd/server/api/helpers.go]: @deads2k what I was thinking about was something like:

```
func IsAPIEnabled(name string, values []string) bool {
  flagValues := sets.NewString(values...)
  return flagValues.Has(name + "=true")
}
```

**Comment by deads2k** [comment]
@mfojtik updated to pass tests.  Take another look through?

File: pkg/cmd/server/api/group_coverage_test.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func TestAllowedAPIVersions(t *testing.T) {
 			enabled.Insert(enabledVersion.Version)
 		}
 		expected := sets.NewString(versions...)
-		actual := enabled.Difference(sets.NewString(api.KubeAPIGroupsToDeadVersions[group]...))
+		actual := enabled.Difference(sets.NewString(api.KubeDefaultDisabledVersions[group]...))
```
Comment (mfojtik) [inline pkg/cmd/server/api/group_coverage_test.go]: why this change?

File: pkg/cmd/server/api/types.go [hunk 3]
```diff
@@ -107,15 +107,16 @@ var (
 		OriginAPIGroupSecurity:      {"v1"},
 	}
 
-	// Map of group names to known, but disallowed REST API versions
-	KubeAPIGroupsToDeadVersions = map[string][]string{
+	// Map of group names to known, but disabled REST API versions
+	KubeDefaultDisabledVersions = map[string][]string{
```
Comment (mfojtik) [inline pkg/cmd/server/api/types.go]: nvmd first comment.

**Comment by mfojtik** [comment]
still LGTM

**Comment by smarterclayton** [comment]
LGTM

**Comment by deads2k** [comment]
[merge]

**Comment by mfojtik** [comment]
@deads2k rebase :(

**Comment by kargakis** [comment]
Note sure if this real or a flake:
<details>

--- FAIL: TestNewWithDelegate (0.03s)
	config_test.go:134: HTTP/1.1 200 OK
		Content-Length: 286
		Content-Type: application/json
		Date: Mon, 03 Jul 2017 23:59:06 GMT
		
		{
		  "paths": [
		    "/apis",
		    "/bar",
		    "/foo",
		    "/healthz",
		    "/healthz/delegate-health",
		    "/healthz/ping",
		    "/healthz/poststarthook/delegate-post-start-hook",
		    "/healthz/poststarthook/wrapping-post-start-hook",
		    "/healthz/wrapping-health",
		    "/swaggerapi"
		  ]
		}
	config_test.go:134: HTTP/1.1 500 Internal Server Error
		Content-Length: 229
		Content-Type: text/plain; charset=utf-8
		Date: Mon, 03 Jul 2017 23:59:06 GMT
		X-Content-Type-Options: nosniff
		
		[+]ping ok
		[-]wrapping-health failed: reason withheld
		[-]delegate-health failed: reason withheld
		[+]poststarthook/delegate-post-start-hook ok
		[-]poststarthook/wrapping-post-start-hook failed: reason withheld
		healthz check failed
		
	config_test.go:142: "http://127.0.0.1:45803/healthz" expected [+]ping ok
		[-]wrapping-health failed: reason withheld
		[-]delegate-health failed: reason withheld
		[+]poststarthook/delegate-post-start-hook ok
		[+]poststarthook/wrapping-post-start-hook ok
		healthz check failed
		, got [+]ping ok
		[-]wrapping-health failed: reason withheld
		[-]delegate-health failed: reason withheld
		[+]poststarthook/delegate-post-start-hook ok
		[-]poststarthook/wrapping-post-start-hook failed: reason withheld
		healthz check failed
	config_test.go:134: HTTP/1.1 500 Internal Server Error
		Content-Length: 51
		Content-Type: text/plain; charset=utf-8
		Date: Mon, 03 Jul 2017 23:59:06 GMT
		X-Content-Type-Options: nosniff
		
		internal server error: delegate failed healthcheck
		
	config_test.go:134: HTTP/1.1 500 Internal Server Error
		Content-Length: 51
		Content-Type: text/plain; charset=utf-8
		Date: Mon, 03 Jul 2017 23:59:06 GMT
		X-Content-Type-Options: nosniff
		
		internal server error: wrapping failed healthcheck
		
	config_test.go:134: HTTP/1.1 200 OK
		Content-Length: 2
		Content-Type: text/plain; charset=utf-8
		Date: Mon, 03 Jul 2017 23:59:06 GMT
		
		ok
	config_test.go:134: HTTP/1.1 500 Internal Server Error
		Content-Length: 36
		Content-Type: text/plain; charset=utf-8
		Date: Mon, 03 Jul 2017 23:59:06 GMT
		X-Content-Type-Options: nosniff
		
		internal server error: not finished
		
	config_test.go:142: "http://127.0.0.1:45803/healthz/poststarthook/wrapping-post-start-hook" expected ok, got internal server error: not finished
	config_test.go:145: "http://127.0.0.1:45803/healthz/poststarthook/wrapping-post-start-hook" expected 200, got 500
	config_test.go:134: HTTP/1.1 403 Forbidden
		Content-Type: text/plain; charset=utf-8
		Date: Mon, 03 Jul 2017 23:59:06 GMT
		Content-Length: 0
		
		
	config_test.go:134: HTTP/1.1 401 Unauthorized
		Content-Type: text/plain; charset=utf-8
		Date: Mon, 03 Jul 2017 23:59:06 GMT
		Content-Length: 0
</details>

**Comment by deads2k** [comment]
re[test] re[merge]
